src/nplinker/genomics/bgc.py

```diff
@@ -201,6 +203,36 @@ def to_dict(self) -> dict[str, Any]:
             "antismash_region": self.antismash_region,
         }
 
+    @staticmethod
+    def to_string(value: Any) -> str:
+        """Convert various types of values to a string.
+
+        Args:
+            value: The value to be converted to a string.
+                Can be a list, tuple, set, dict, or any other type.
+
+        Returns:
+            A string representation of the input value.
+        """
+        # Convert list, tuple, set to comma-separated string
+        if isinstance(value, (list, tuple, set)):
+            value = ", ".join(map(str, value))
+        # Convert dict to comma-separated string
+        elif isinstance(value, dict):
+            value = ", ".join([f"{k}:{v}" for k, v in value.items()])
+        # Convert anything else to string
```

Comment:
Do you really want to covert `None` to ""? Does it make sense to the BGC attributes?

Reply:
If the value is `None`, the corresponding field in the tabular output file should be left empty. This ensures that when the file is opened in Excel, numeric fields are correctly recognized as numbers rather than text, allowing the columns to be sorted properly. For text fields, leaving them empty is also preferable to displaying None, as it is cleaner and more intuitive.